src/build/app/mod.rs

```diff
@@ -675,9 +676,36 @@ impl<'b> App<'b> {
         for arg in args.into_iter() {
             self.args.push(arg.into());
         }
+        self.panic_on_missing_help();
```

Comment:
I don't think it should be called here. We have something called a global setting which can be propagated from top to below during build of the app. Check for `_build()` in `App`.

If you write a test where you add an arg before adding a setting, it wouldn't panic.

Reply:
Ugh, *this* is why I requested to check subcommands explicitly! They were suppose to write the test, figure that it doesn't work, go through WTF stage, eureka!

Now, part of this educational effect is lost :(

Reply:
Thanks, that is true, indeed. I will look into this tomorrow!

Reply:
Thanks for your help, both @CreepySkeleton  and @pksunkara.
- I added a test where you add an arg before adding a setting.
- I moved the panic_on_missing_help() into the _build() method

**Problem:** Now, I will only get a panic, if _build actually gets called. So this will panic:

`#[test]
#[should_panic]
fn help_required_but_not_given() {
    App::new("myapp")
        .setting(AppSettings::HelpRequired)
        .arg(Arg::with_name("foo"))
        .get_matches();
}`

but this will not 

`#[test]
#[should_panic]
fn help_required_but_not_given() {
    App::new("myapp")
        .setting(AppSettings::HelpRequired)
        .arg(Arg::with_name("foo"));
}`

Is this acceptable?

Reply:
Yup. Just hide the `.get_matches()` from docs as per my earlier comment.

Reply:
Ah, I misunderstood your earlier comment then. I thought I should delete it.
But do you think it is a good idea to show a "panic"-example in the docs, which does not actually panic?

Reply:
That's okay because we are talking about the AppSetting during building only.

Reply:
Okay, I implemented it, like you suggested in the last commit.